<!--
  Copyright (C) 2018-2021 LEIDOS.
  Licensed under the Apache License, Version 2.0 (the "License"); you may not
  use this file except in compliance with the License. You may obtain a copy of
  the License at
  http://www.apache.org/licenses/LICENSE-2.0
  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
  License for the specific language governing permissions and limitations under
  the License.
-->
<!--
  This file is used to launch a truck inspection client node.
-->
<launch>
    <arg name="truck_inspection_client_param_file" default="$(find truck_inspection_client)/config/parameters.yaml" />

    <remap from="outgoing_mobility_operation" to="$(optenv CARMA_MSG_NS)/outgoing_mobility_operation"/>
    <remap from="bsm_outbound" to="$(optenv CARMA_MSG_NS)/bsm_outbound"/>
    <remap from="incoming_mobility_request" to="$(optenv CARMA_MSG_NS)/incoming_mobility_request"/>
    <remap from="guidance/state" to="$(optenv CARMA_GUIDE_NS)/state"/>

    <node name="truck_inspection_client" pkg="truck_inspection_client" type="truck_inspection_client">
      <rosparam command="load" file="$(arg truck_inspection_client_param_file)"/>
    </node>
</launch>
